### DISCUSSION TITLE
Which Queued Tasks are currently running?

### DISCUSSION THREAD

**Discussion by Tobi-De** [django-q2/django-q2#143] [Q&A] [answered]
Is there a way to know the currently running tasks, regardless of the broker being used? For example, with Redis, you don't see queued tasks in the admin. Even if that were the case, it may not be a solution since (not certain, could be incorrect) a queue in the admin does not always correspond to an actively running task; it might be a task waiting to be run but not actively running. What I want to know is if there is a way to know when a task (using it name for example) is actively running.
My use case is making some kind of singleton task, I have a task that can be triggered by multiple clients at the same time, but I need to have only one instance of that task running at a time.

**Comment by GDay** [comment, marked as answer]
This is currently not tracked. 

The easiest way to do this right now is to track it yourself. For example, by putting something in cache to indicate that the task started and then in the other tasks add a check on that before continuing with the task.

**Reply by Tobi-De** [reply]
That's what I though, thanks.
